vulture_os/toolkit/api_parser/signalsciences_ngwaf/signalsciences_ngwaf.py

```diff
@@ -0,0 +1,226 @@
+        'Accept': 'application/json'
+    }
+
+    def __init__(self, data):
+        super().__init__(data)
+
+        self.signalsciences_ngwaf_email = data["signalsciences_ngwaf_email"]
+        self.signalsciences_ngwaf_token = data["signalsciences_ngwaf_token"]
+        self.signalsciences_ngwaf_corp_name = data["signalsciences_ngwaf_corp_name"]
+        self.signalsciences_ngwaf_site_name = data["signalsciences_ngwaf_site_name"]
+
+        self.session = None
+
+
+    def _connect(self):
+        try:
+            if self.session is None:
+                self.session = requests.Session()
+
+                headers = self.HEADERS
+                headers.update({
+                    'x-api-user': self.signalsciences_ngwaf_email,
+                    'x-api-token': self.signalsciences_ngwaf_token
+                })
+
+                self.session.headers.update(headers)
+
+            return True
+
+        except Exception as e:
+            raise SignalSciencesNgwafAPIError(f"{[__parser__]}:connect: An error occurred when creating http session : {str(e)}")
+
+
+    def __execute_query(self, url, query=None, timeout=10):
+        try:
+            self._connect()
+
+            response = self.session.get(
+                url,
+                params=query,
+                timeout=timeout,
+                proxies=self.proxies,
+                verify=self.api_parser_custom_certificate if self.api_parser_custom_certificate else self.api_parser_verify_ssl)
+
+            if response.status_code != 200:
+                raise SignalSciencesNgwafAPIError(
+                    f"Error at SafenetAPI Call URL: {url} Code: {response.status_code} Content: {response.content}")
+
+            return response.json()
+
+        except Exception as e:
+            logger.exception(f"{[__parser__]}:__execute_query: {str(e)}", extra={'frontend': str(self.frontend)})
+            raise SignalSciencesNgwafAPIError(f"{[__parser__]}:__execute_query: An error occurred while executing query : {str(e)}")
+
+
+    def test(self):
+        current_time = timezone.now().replace(second=0, microsecond=0)
+        try:
+            logs = self.get_logs(
+                since=current_time - timedelta(minutes=6),
+                to=current_time - timedelta(minutes=5))
+
+            return {
+                "status": True,
+                "data": logs
+            }
+        except Exception as e:
+            logger.exception(f"{[__parser__]}:test: {str(e)}", extra={'frontend': str(self.frontend)})
+            return {
+                "status": False,
+                "error": str(e)
+            }
+
+
+    def get_logs(self, since=None, to=None):
+        try:
+            url = self.API_BASE_URL + '/api/v0/corps/%s/sites/%s/feed/requests'%(self.signalsciences_ngwaf_corp_name, self.signalsciences_ngwaf_site_name)
+
+            query = {}
+            if isinstance(since, datetime):
+                query['from'] = int(since.timestamp() / 60) * 60
```

Comment:
I would prefer doing that kind of calculation/restraint in the execution phase, to store correct since/to values in the `last_api_call`

Reply:
this is still not necessary if `since` and `to` values are correctly set when calling the function, and it should